tests/unit/package_managers/test_generic.py

```diff
@@ -1,26 +1,62 @@
+from typing import Type
 from unittest import mock
 
 import pytest
 
-from cachi2.core.errors import PackageRejected
+from cachi2.core.errors import Cachi2Error, PackageManagerError, PackageRejected
 from cachi2.core.models.input import GenericPackageInput
 from cachi2.core.models.sbom import Component
-from cachi2.core.package_managers.generic import (
+from cachi2.core.package_managers.generic.main import (
     DEFAULT_LOCKFILE_NAME,
+    _load_lockfile,
     _resolve_generic_lockfile,
     fetch_generic_source,
 )
+from cachi2.core.package_managers.generic.models import GenericLockfileV1
 from cachi2.core.rooted_path import RootedPath
 
+LOCKFILE_WRONG_VERSION = """
+metadata:
+    version: '0.42'
+artifacts:
+    - download_url: https://redhat.com/artifact
+      checksums:
+        md5: 3a18656e1cea70504b905836dee14db0
+"""
+
+LOCKFILE_CHECKSUM_MISSING = """
+metadata:
+    version: '1.0'
+artifacts:
+    - download_url: https://redhat.com/artifact
```

Comment:
nitpick: Since the URLs are pretty much just placeholders in unit tests, let's stay "unaffiliated" and try to come up with some 'foobarized' vendor-agnostic version of it :) (there should be other examples in the code base already)